.github/workflows/dart_ci.yml

```diff
@@ -0,0 +1,90 @@
+name: Dart CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        sdk: [stable, dev]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dart-lang/setup-dart@v0.1
+        with:
+          channel: ${{ matrix.sdk }}
+
+      - name: Print Dart SDK version
+        run: dart --version
+
+      - id: install
+        name: Install dependencies
+        run: dart pub get
+
+      - name: Validate dependencies
+        run: dart pub run dependency_validator --no-fatal-pins -i analyzer,build_runner,build_web_compilers,built_value_generator
+        if: always() && steps.install.outcome == 'success'
+
+      - name: Verify formatting
+        run: dart pub run dart_dev format --check
+        if: always() && ${{ matrix.sdk }} == 'stable' && steps.install.outcome == 'success'
+
+      # Analyze before generated files are created to verify that component boilerplate analysis is "clean" without the need for building
```

Comment:
👏 

This is great, but since we check in generated files in this repo, this doesn't have the desired effect. 

In a separate PR, though (which I've been wanting to do for a while), we can stop checking in files outside of lib, so that components within example and test directories do actually get analyzed without generated code.